srcjs/utils.js

```diff
@@ -240,7 +240,12 @@ function mergeSort(list, sortfunc) {
 
 // Escape jQuery selector metacharacters: !"#$%&'()*+,./:;<=>?@[\]^`{|}~
 var $escape = exports.$escape = function(val) {
-  return val.replace(/([!"#$%&'()*+,.\/:;<=>?@\[\\\]^`{|}~])/g, '\\$1');
+  // Sometime val comes empty, and Js throws a 'e.replace() is not a function' error
```

Comment:
@wch when running the current version of `{shiny}` with the example app at #2688, there is a JavaScript error: 

> shiny.min.js:3 Uncaught TypeError: e.replace is not a function

![Screenshot 2020-09-11 at 08 42 12](https://user-images.githubusercontent.com/17936236/92876771-c4649d80-f40a-11ea-968b-986a2653cbda.png)

When adding `console.log(val)` in `$escape`, I found that when running `updateRadioButtons()`, `val` is `[]`, hence throwing an error here.

Reply:
Are you sure this is needed now that falsy values are properly accounted for in `.setValue()` (note that `updateRadioButton()` calls `receiveMessage()` which then calls `setValue()`)? Seems to me like it isn't (but I haven't verified)

Reply:
I guess it makes sense that a `[]` is sent, since that's what `shiny:::toJSON(character(0))` returns.

That said, I still think it makes sense for that `$escape` to keep the original behavior. And I believe that with the new logic you've added for `setValue`, it will no longer call `$escape([])`, so the error would not happen even with the original version of `$escape`.